radicale/storage.py

```diff
@@ -474,3 +517,61 @@ def serialize(self):
         elif self.get_meta("tag") == "VADDRESSBOOK":
             return "".join([item.serialize() for item in items])
         return ""
+
+    _lock = threading.Lock()
+
+    @classmethod
+    def acquire_lock(cls, mode):
```

Comment:
You might want to experiment with making this entire method just a contextmanager, to avoid having to return a stub Lock instance.